lib/bai2/integrity.rb

```diff
@@ -77,33 +79,39 @@ def assert_integrity!
         end
 
         # Run children assertions, which return number of records. May raise.
-        records = self.accounts.map {|a| a.send(:assert_integrity!) }.reduce(0, &:+)
+        records = self.accounts.map {|a| a.send(:assert_integrity!, options) }.reduce(0, &:+)
 
-        unless expectation[:records] == (actual = records + 2)
+        unless expectation[:records] == (actual_num_records = records + 2)
           raise IntegrityError.new(
-            "Record count invalid: group: #{expectation[:records]}, accounts: #{actual}")
+            "Record count invalid: group: #{expectation[:records]}, accounts: #{actual_num_records}")
         end
 
         # Return record count
-        records + 2
+        actual_num_records
       end
     end
 
 
     class Account
       private
 
-      def assert_integrity!
+      def assert_integrity!(options)
         expectation = {
           sum:      @trailer[:account_control_total],
           records:  @trailer[:number_of_records],
         }
 
         # Check sum vs. summary + transaction sums
-        actual_sum = self.transactions.map(&:amount).reduce(0, &:+) \
-          #+ self.summaries.map {|s| s[:amount] }.reduce(0, &:+)
-          # TODO: ^ there seems to be a disconnect between what the spec defines
-          # as the formula for the checksum and what SVB implements...
+        summary_amounts_sum = self.summaries.map {|s| s[:amount] }.reduce(0, &:+)
+        transaction_amounts_sum = self.transactions.map(&:amount).reduce(0, &:+)
+
+        # Some banks differ from the spec (*cough* SVB) and do not include
+        # the summary amounts in the control amount.
+        actual_sum = if options[:account_control_ignores_summary_amounts]
+                       transaction_amounts_sum
+                     else
+                       transaction_amounts_sum + summary_amounts_sum
+                     end
```

Comment:
This is what makes it a breaking change (and why I bumped it to 2.0.0). I agree with the comment here: https://github.com/venturehacks/bai2/commit/f1601dbd4f2a7041d8bc2939a0ef203c33229fb8 (We should validate by default) 

People have forked this library just to change the old behavior. See: https://github.com/net-engine/bai2/commit/0c5fcae9a5ac04a752e26a908ca33236840b9b66

Reply:
:+1: Let's add some release notes as well, since apparently people are using this. We could send a message to the people who've forked as well if you'd like to let them know.